File: Leetcode/328-Odd-Even-Linked-List/OddEvenLinkedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {
        if (head == NULL) {
            return head;
        }

        ListNode* o = head;
        ListNode* e = head->next;
        ListNode* eHead = head->next;
        while (e != NULL && e->next != NULL) {
            o->next = e->next;
            e->next = e->next->next;
            o = o->next;
            e = e->next;
        }
        o->next = eHead;
        return head;
    }
};

File: Leetcode/328-Odd-Even-Linked-List/OddEvenLinkedList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var oddEvenList = function(head) {
    if (!head) {
        return head
    }

    var o = head
    var e = head.next
    var eHead = head.next
    while (e && e.next) {
        o.next = e.next
        e.next = e.next.next
        o = o.next
        e = e.next
    }
    o.next = eHead
    return head
};
